app/src/backend/CalculationService.ts

```diff
@@ -73,70 +81,193 @@ export default class CalculationService {
-            );
-          }
-          // TODO save amount to GasValue entry?
-          const amount = BigInt(data[key]);
-          const { co2eq, co2eqYears } = this.calculateCO2eq(
-            gasToCO2Eqs,
-            gasName,
-            amount,
-          );
-          totalCO2e += co2eq;
-          totalCO2eYears = Math.max(totalCO2eYears, co2eqYears);
-          return { gas: gasName, amount: amount };
-        });
+        gases = CalculationService.handleDirectMeasureFormula(
+          activityValue,
+          gasToCO2Eqs,
+        );
         break;
       case "activity-amount-times-emissions-factor":
-        // TODO add actvityAmount column to ActivityValue
-        // const activityAmount = activityValue.activityAmount || 0;
-        const data = activityValue.activityData;
-        const activityAmount = data ? data["activity_amount"] || 0 : 0;
-        gases = activityValue.gasValues.map((gasValue) => {
-          const emissionsFactor = gasValue.emissionsFactor;
-          if (emissionsFactor == null) {
-            throw new createHttpError.BadRequest(
-              "Missing emissions factor for activity",
-            );
-          }
-          if (emissionsFactor.emissionsPerActivity == null) {
-            throw new createHttpError.BadRequest(
-              `Emissions factor ${emissionsFactor.id} has no emissions per activity`,
-            );
-          }
-          // this rounds/ truncates!
-          const amount = BigInt(
-            activityAmount * emissionsFactor.emissionsPerActivity,
-          );
-          const { co2eq, co2eqYears } = this.calculateCO2eq(
+        gases =
+          CalculationService.handleActivityAmountTimesEmissionsFactorFormula(
+            activityValue,
             gasToCO2Eqs,
-            gasValue.gas!,
-            amount,
           );
-          totalCO2e += co2eq;
-          totalCO2eYears = Math.max(totalCO2eYears, co2eqYears);
-
-          return { gas: gasValue.gas!, amount };
-        });
         break;
+      case "methane-commitment":
+        gases = CalculationService.handleMethaneCommitmentFormula(
+          activityValue,
+          gasToCO2Eqs,
+        );
+        break;
+      case "induced-activity-1": // TODO or VKT-1? What is it in the hierarchy.json file?
+        gases = CalculationService.handleVkt1Formula(
+          activityValue,
+          gasToCO2Eqs,
+        );
       default:
         throw new createHttpError.NotImplemented(
           `Formula ${formula} not yet implemented for input methodology ${inventoryValue.inputMethodology}`,
         );
     }
+
+    for (const gas of gases) {
+      const { co2eq, co2eqYears } = this.calculateCO2eq(
+        gasToCO2Eqs,
+        gas.gas,
+        gas.amount,
+      );
+      totalCO2e += co2eq;
+      totalCO2eYears = Math.max(co2eqYears, totalCO2eYears);
+    }
+
     return {
       totalCO2e,
       totalCO2eYears,
       gases,
     };
   }
+
+  static handleVkt1Formula(
```

Comment:
Can we move the formulas to another class and have the calculation service depend on this class. We will be implementing a lot of forumulas in the coming weeks, which means the file will get really large

Reply:
Fair, that's why I originally created the CalculationService but yes it makes sense to split the public methods from the private implementation details. So I guess we can either have a FormulaService, or we can just have a formulas.ts where these are exported methods?